server.go

```diff
@@ -298,16 +301,41 @@ func removeTagsHTTP(w http.ResponseWriter, r *http.Request) {
 
 // Set the target file's name
 func setImageNameHTTP(w http.ResponseWriter, r *http.Request) {
-	bytes, err := ioutil.ReadAll(r.Body)
-
+	err := r.ParseForm()
 	if err != nil {
 		sendError(w, 400, err)
-	} else if len(bytes) > 200 {
-		bytes = bytes[0:200]
 	}
 
-	err = setImageName(extractParam(r, "id"), string(bytes))
+	name := r.Form.Get("name")
+	if len(name) > 200 {
+		name = name[0:200]
+	}
+
+	err = setImageName(extractParam(r, "id"), name)
+
+	switch err {
+	case nil:
+	case sql.ErrNoRows:
+		sendError(w, 400, err)
+	default:
+		send500(w, r, err)
+	}
+}
+
+// Fetch tags for a single file
+func fetchTagsHTTP(w http.ResponseWriter, r *http.Request) {
+	sha1 := extractParam(r, "id")
+	pair, err := db.GetImageIDAndMD5(sha1)
+	if err != nil {
+		send500(w, r, err)
+	}
+
+	tags, err := fetch.FetchTags(pair.MD5)
+	if err != nil {
+		send500(w, r, err)
```

Comment:
And here. Yes, we need a better error handling scheme. 
```suggestion
		send500(w, r, err)
		return
```